utilities/virt.py

```diff
@@ -2632,3 +2636,64 @@ def username_password_from_cloud_init(vm_volumes: list[dict[str, Any]]) -> tuple
             return _match["user"], _match["password"]
 
     return "", ""
+
+
+def validate_virtctl_guest_agent_after_guest_reboot(vm: VirtualMachineForTests, os_type: str) -> bool:
```

Comment:
if it is used only inside 'tests/virt/cluster' - you can keep it there, no need to move to common place

also, this function contains too many actions (reboot, wait when running, validate ga), please split it and rename to something like "reboot_guest_and_wait_for_running", it should include only steps `guest_reboot` and `wait_for_running_vm`

and it should be used as a fixture in all tests

in the test body you should call old function: `validate_virtctl_guest_agent_data_over_time`